bootstrap/src/test.rs

```diff
@@ -126,25 +191,28 @@ impl FileChecker {
         Self { filecheck }
     }
 
-    fn run(&self, source: &Path, output: &Path) {
-        let case = source.file_stem().unwrap().to_string_lossy();
-        let generated = std::fs::read_dir(output.parent().unwrap())
-            .unwrap()
-            .filter_map(|entry| entry.ok())
-            .find(|entry| {
-                let filename = entry.file_name();
-                let filename = filename.to_string_lossy();
-                filename.ends_with(".c") && filename.starts_with(case.as_ref())
-            });
-
-        assert!(generated.is_some(), "could not find {case}'s generated file");
-        let generated = generated.unwrap();
-
-        let generated = File::open(generated.path()).unwrap();
+    fn run(&self, case: &TestCase) {
+        let generated = File::open(case.generated()).unwrap();
         let mut command = std::process::Command::new(&self.filecheck);
-        command.arg(source).stdin(generated);
+        command.arg(&case.source).stdin(generated);
         log::debug!("running {:?}", command);
         let output = command.output().unwrap();
-        assert!(output.status.success(), "failed to run FileCheck on {case}");
+        assert!(
+            output.status.success(),
+            "failed to run FileCheck on {}",
+            case.source.file_stem().unwrap().to_string_lossy()
+        );
+    }
+}
+
+fn bless(update: bool, case: &TestCase) {
+    let output = case.generated();
+    let blessed = case.source.with_extension("c");
+    if update {
+        std::fs::copy(output, blessed).unwrap();
+    } else {
+        let output = std::fs::read(output).unwrap();
+        let blessed = std::fs::read(blessed).unwrap();
+        assert_eq!(output, blessed, "output does not match blessed output");
```

Comment:
Could this use something like [`similar`](https://docs.rs/similar/2.6.0/similar/) for better output?